### ISSUE TITLE
SSL CERT

### ISSUE THREAD

**Issue by costelabr17** [CaramelDunes/prodinfo_gen#15]
is possible grab my donor prodinfo to gen a donor with ssl certificate? i have a banned switch and brought a new one today damaged but i can recover the backup nand

**Comment by CaramelDunes** [comment]
You mean using a donor, not banned, console to connect to Nintendo online services?
Importing the SSL certificate can be done but is not enough/will probably get you banned.

I think using a donor prodinfo as-is and spoofing the device ID should work, but I haven't been able to try.
See https://gbatemp.net/threads/use-prodinfo-from-another-switch.581963/ for more info.

**Comment by costelabr17** [comment]
Yes i know about this de ice spoof, but need atmosphere for boot console using prodinfo_gen no need console boot normaly, that is why i want to import ssl cert, is possible?

**Comment by CaramelDunes** [comment]
Importing the SSL cert and other certificates is possible. You'd probably get banned, though, as there would be a mismatch between the console device id and the device id embedded in the certificates.

**Comment by Kronos2308** [comment]
thats true for example if you spoof the ssl certificate from the switch with charles proxy you h¡get the token but the deviv¡cce id mitsmatch so no  game downloading the shop chash and no online games

**Comment by CaramelDunes** [comment]
> 
> 
> thats true for example if you spoof the ssl certificate from the switch with charles proxy you get the token but the device id mitsmatch so no game downloading the shop chash and no online games

That's good info, thanks for sharing.

**Comment by costelabr17** [comment]
You can explain how i can build my own version of prodinfo_gen payload to gen a donor_prodinfo without wipe ssl cert?

**Comment by costelabr17** [comment]
or i can do it manually in HxD but i don't the exactly strings to copy

**Comment by Kronos2308** [comment]
is beter not use adonor ssl cert i won't work and you will get banned

**Comment by Kronos2308** [comment]
if you want to usea donnor prodinfo you have to used the raw unecripted prodinfo 
and spoff the device id the match with that prodinfo to boot

**Comment by CaramelDunes** [comment]
Thanks for stepping in @Kronos2308 👍 
He's right, @costelabr17. Also, if your prodinfo is personalized (non-Erista), you need special key blobs to be able to decrypt the SSL private key and those are not on the internet AFAIK.

**Comment by Kronos2308** [comment]
Yep
On a normal switch the OS take the device id from the board and from the prodinfo
And compare both if are the same continue if not the get stuck on switch logo
On a CFW switch Atmosphere exosphere take the device id from the board and from the prodinfo

In order to use a different prodinfo
You mush make a backup XD

You must take another uncrypted prodinfo and encrypted with your BIS key0 and writhe that to the Prodinfo partition to the switch
Now you have a problem because the console won't boot , that's because the device id not match
So you have two options 
* you change the device id in theprodinfo before writing
(@CaramelDunes  there is a way of just change the device id?)
* or you spoof the device id using atmosphere exosphere. Since atmosphere not allow that you have to make that your self and use a custom exosphere that way the console boot but only if there are in CFW
Both device id has to match

**Comment by CaramelDunes** [comment]
>     * you change the device id in theprodinfo before writing
>       (@CaramelDunes  there is a way of just change the device id?)

You can, indeed, modify the prodinfo enough that it boots (and that's basically what `prodinfo_gen` does), but this also corrupts some of the keys that the Switch uses to connect to online services.

**Comment by Kronos2308** [comment]
but i mean to just mod the part that store the device id

**Comment by CaramelDunes** [comment]
Unfortunately there's no "one" part that stores the device ID. It is embedded in a lot of parts, and most of those are digitally signed; modifying them corrupt them.

**Comment by Kronos2308** [comment]
![](https://cdn.discordapp.com/emojis/703059890222006352.png?size=48&size=64 )
well then the exosphere part then 
BTW
the exoshere id can be taken from pridinfo with ams mitm
so if you can make a work arround you can spoof and boot any prodinfo just with ams withour touch the Prodinfo partition
is posiblle qith the actual ams but must be implemented
